url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/pl/cc/SystemCoreState.java

```diff
@@ -0,0 +1,685 @@
+import pl.cc.core.cmd.EventAgentQueueMember;
+import pl.cc.core.cmd.EventAgentRemoveFromQueue;
+import pl.cc.core.cmd.EventAgentUnPause;
+import pl.cc.core.cmd.EventQueueJoin;
+import pl.cc.core.cmd.EventQueueLeave;
+import pl.cc.core.cmd.InfoAgent;
+import pl.cc.core.cmd.InfoPauseDefinition;
+import pl.cc.core.cmd.InfoPrivilege;
+import pl.cc.core.cmd.InfoQueue;
+import pl.cc.core.cmd.dummy.Loggedout;
+import pl.cc.exceptions.AgentNotFoundException;
+import pl.cc.exceptions.CallNotFoundException;
+import pl.cc.exceptions.QueueNotFoundException;
+import pl.cc.real.RealAgent;
+import pl.cc.real.RealAgentList;
+import pl.cc.real.RealCall;
+import pl.cc.real.RealCallList;
+import pl.cc.real.RealQueue;
+import pl.cc.real.RealQueueList;
+
+/**
+ * Budujemy stan wnętrza systemu na podstawie komunikatów płynących z CCProxy
+ * - Lista zalogowanych agentów
+ * - Lista kolejek w systemie
+ * 
+ * @since 2009-05-21
+ */
+public class SystemCoreState {
+	public static final Logger log = Logger.getLogger(SystemCoreState.class);
+
+	private RealQueueList qList = new RealQueueList();
+	private RealAgentList aList = new RealAgentList();
+	private RealCallList callList = new RealCallList();
+	
+	/**
+	 * Lista agentów do których mamy uprawnienia
+	 */
+	private RealAgentList privAgentList = new RealAgentList();
+	/**
+	 * Lista zdefiniowanych pauz w systemie. Lista spływa z CCProxy
+	 */
+	private PauseTypeList pauseTypeList = new PauseTypeList();
+	
+	public SystemCoreState(){
+		log.info("======== Creating new instance SystemCoreState ========");
+		// FIXME: TESTY - testy listenerów dla CallList
+		/*
+		RealCallListObserver observer = new RealCallListObserver("GLOBAL");
+		callList.addCallListListener(observer);
+		
+		qList.addQueueListListener(new RealCallListObserver("BLUZECZKA"));
+		*/
+		// koniec testów
+	}
+
+	public SystemCoreState(PauseTypeList pauseTypeList){
+		this();
+		this.pauseTypeList = pauseTypeList; 
+	}
+
+	
+	/**
+	 * @return Lista agentów zalogowanych do systemu
+	 */
+	public RealAgentList getLoggedAgentList(){
+		return aList;
+	}
+	/**
+	 * @return Lista kolejek w systemie
+	 */
+	public RealQueueList getQueueList(){
+		return qList;
+	}
+
+	/**
+	 * @return Poł
+	 */
+	public RealCallList getCallList(){
+		return callList;
+	}
+
+	/**
+	 * @return Lista defninicji pauz używanych w systemie
+	 */
+	PauseTypeList getPauseTypeList() {
+		return pauseTypeList;
+	}
+	/**
+	 * @param agent Czy mamy uprawnienie do wybranego agenta
+	 * @return true jeśli mamy uprawnienie
+	 */
+	public boolean hasPrivilegeForAgent(RealAgent agent){
+		return privAgentList.containsValue(agent);
+	}
+	
+	/**
+	 * @param cmd - Komenda która napłynęła z CCProxy
+	 * @param pt ProxyTasks - namiastka połączenia do CCProxy - czasem agenci sami inicjują jakieś akcje w CCProxy (np kliknięcię Spy)
+	 * @return true - jeśli komenda została faktycznie przetworzona, false jeśli nie byliśmy nią zainteresowani
+	 */
+	public boolean process(Command cmd, ProxyTasks pt){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 50. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.